### PR TITLE
FLINK-4482 numUnsuccessfulCheckpointsTriggers is accessed without holding triggerLock

### PR DESCRIPTION
**Pull request by tedyu**
Thanks for contributing to Apache Flink. Before you open your pull request, please take the following check list into consideration.
If your changes take all of the items into account, feel free to open your pull request. For more information and/or questions please refer to the [How To Contribute guide](http://flink.apache.org/how-to-contribute.html).
In addition to going through the list, please provide a meaningful description of your changes.

- [ ] General
  - The pull request references the related JIRA issue ("[FLINK-XXX] Jira title text")
  - The pull request addresses only one issue
  - Each commit in the PR has a meaningful commit message (including the JIRA id)

- [ ] Documentation
  - Documentation has been added for new functionality
  - Old documentation affected by the pull request has been updated
  - JavaDoc for public methods has been added

- [ ] Tests & Build
  - Functionality added by the pull request is covered by tests
  - `mvn clean verify` has been executed successfully locally or a Travis build has passed

### CODE REVIEW

**Comment by StephanEwen** [comment]
I think it would be good to use the `triggerLock` only within the `triggerCheckpoint(...)` method.

The access to `numUnsuccessfulCheckpointsTriggers` is not super critical, it is merely an informational variable. We could make it an AtomicInteger, if that helps.

**Comment by tedyu** [comment]
The test failure in flink-mesos is unrelated to the PR.

File: flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java [hunk 1]
```diff
@@ -135,7 +136,7 @@
 	private JobStatusListener jobStatusListener;
 
 	/** The number of consecutive failed trigger attempts */
-	private int numUnsuccessfulCheckpointsTriggers;
+	private AtomicInteger numUnsuccessfulCheckpointsTriggers = new AtomicInteger(0);
```
Comment (StephanEwen) [inline flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java]: This should be final

File: flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java [hunk 2]
```diff
@@ -401,7 +402,7 @@ CheckpointTriggerResult triggerCheckpoint(long timestamp, CheckpointProperties p
 				checkpointID = checkpointIdCounter.getAndIncrement();
 			}
 			catch (Throwable t) {
-				int numUnsuccessful = ++numUnsuccessfulCheckpointsTriggers;
+				int numUnsuccessful = numUnsuccessfulCheckpointsTriggers.addAndGet(1);
```
Comment (StephanEwen) [inline flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java]: Minor: `incrementAndGet()` reads nice here.

File: flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java [hunk 3]
```diff
@@ -503,7 +504,7 @@ else if (!props.isSavepoint()) {
 					pendingCheckpoints.remove(checkpointID);
 				}
 
-				int numUnsuccessful = ++numUnsuccessfulCheckpointsTriggers;
+				int numUnsuccessful = numUnsuccessfulCheckpointsTriggers.addAndGet(1);
```
Comment (StephanEwen) [inline flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java]: same as above

**Comment by StephanEwen** [comment]
Two minor comments, otherwise +1

**Comment by StephanEwen** [comment]
Merging this...

**Pull request closed without merging**